From sjg@crufty.net  Wed Oct  4 00:46:22 2017
From: Simon Gerraty <sjg@crufty.net>
To: Steffen Nurpmeso <steffen@sdaoden.eu>
Subject: Re: bmake: errors when creating Alpine Linux/Musl package
MIME-Version: 1.0
Content-Type: text/plain; charset="us-ascii"
Content-ID: <24543.1507070779.1@bad.crufty.net>
Date: Tue, 03 Oct 2017 15:46:19 -0700
Message-ID: <20363.1507070779@bad.crufty.net>
Status: RO

> |The warning about __inline being redefined - I would not expect
> |missing/sys/cdefs.h to be used on a Linux platform - every one I'd used
> |(not many) has sys/cdefs.h

Not much we can do about that warning.
Fortunately nothing in bmake uses __inline

For the tcc getopt bit, try running strings on the object
to see where that Del char is.

For gcc, try the patch below

Index: unit-tests/dotwait.exp
===================================================================
RCS file: /volume/cvs/bmake/unit-tests/dotwait.exp,v
retrieving revision 1.1.1.1
diff -u -p -r1.1.1.1 dotwait.exp
--- XXX/unit-tests/dotwait.exp	30 Aug 2014 18:57:18 -0000	1.1.1.1
+++ XXX/unit-tests/dotwait.exp	3 Oct 2017 22:43:52 -0000
@@ -22,9 +22,9 @@ shared.2.1
 shared.2.1
 shared.2.99
 shared.2.99
+cycle.1.99
+cycle.1.99
 make: Graph cycles through `cycle.2.99'
 make: Graph cycles through `cycle.2.98'
 make: Graph cycles through `cycle.2.97'
-cycle.1.99
-cycle.1.99
 exit status 0
Index: unit-tests/dotwait.mk
===================================================================
RCS file: /volume/cvs/bmake/unit-tests/dotwait.mk,v
retrieving revision 1.1.1.1
diff -u -p -r1.1.1.1 dotwait.mk
--- XXX/unit-tests/dotwait.mk	30 Aug 2014 18:57:18 -0000	1.1.1.1
+++ XXX/unit-tests/dotwait.mk	3 Oct 2017 22:42:44 -0000
@@ -11,7 +11,7 @@ PAUSE= sleep 1
 # Ignore "--- target ---" lines printed by parallel make.
 all:
 .for t in ${TESTS}
-	@${.MAKE} -f ${THISMAKEFILE} -j4 $t | grep -v "^--- "
+	@${.MAKE} -f ${THISMAKEFILE} -j4 $t 2>&1 | grep -v "^--- "
 .endfor
 
 #

